fiftyone/utils/activitynet.py

```diff
@@ -0,0 +1,1038 @@
+                "a full ActivityNet split, you must download the source files "
+                "from the ActivityNet maintainers. See "
+                "https://voxel51.com/docs/fiftyone/integrations/activitynet.html#lactivitynet-full-split-downloads "
+                "for more information."
+                "\n\n"
+                "Alternatively, provide the `max_samples`, `max_duration`, "
+                "and/or `classes` parameters to download a subset of the "
+                "dataset from YouTube." % (num_existing, num_total, split)
+            )
+
+        num_samples = num_existing
+        sample_ids = None
+    else:
+        (
+            sample_ids,
+            num_downloaded_samples,
+        ) = manager.download_necessary_samples(download_config)
+        num_samples = num_existing + num_downloaded_samples
+
+    manager.write_data_json(split)
+    manager.write_annotations(sample_ids, split)
+
+    return num_samples, all_classes
+
+
+class ActivityNetDatasetImporter(
+    foud.FiftyOneTemporalDetectionDatasetImporter
+):
+    """Class for importing datasets in ActivityNet format.
+
+    See :class:`fiftyone.types.dataset_types.ActivityNetDataset` for format
+    details.
+
+    Args:
+        dataset_dir (None): the dataset directory
+        data_path (None): an optional parameter that enables explicit control
+            over the location of the media. Can be any of the following:
+
+            -   a folder name like ``"data"`` or ``"data"/`` specifying a
+                subfolder of ``dataset_dir`` where the media files reside
+            -   an absolute directory path where the media files reside. In
+                this case, the ``dataset_dir`` has no effect on the location of
+                the data
+            -   a filename like ``"data.json"`` specifying the filename of the
+                JSON data manifest file in ``dataset_dir``
+            -   an absolute filepath specifying the location of the JSON data
+                manifest. In this case, ``dataset_dir`` has no effect on the
+                location of the data
+
+            If None, this parameter will default to whichever of ``data/`` or
+            ``data.json`` exists in the dataset directory
+        labels_path (None): an optional parameter that enables explicit control
+            over the location of the labels. Can be any of the following:
+
+            -   a filename like ``"labels.json"`` specifying the location of
+                the labels in ``dataset_dir``
+            -   an absolute filepath to the labels. In this case,
+                ``dataset_dir`` has no effect on the location of the labels
+
+            If None, the parameter will default to ``labels.json``
+        compute_metadata (False): whether to produce
+            :class:`fiftyone.core.metadata.VideoMetadata` instances for each
+            video when importing
+        shuffle (False): whether to randomly shuffle the order in which the
```

Comment:
I think `ActivityNetDatasetImporter` needs to implement the `classes` and `max_duration` parameters from `download_activitynet_split()`, otherwise if the user downloads a full split but then runs:

```py
dataset = foz.load_zoo_dataset(
    "activitynet-200",
    split="validation",
    classes=["Bathing dog", "Walking the dog"],
    max_duration=20,
)
```

all samples will be loaded, not just the ones matching `classes` and `max_duration` that would be loaded if the same method were being run for the very first time.

This was also true for COCO and OpenImages, where we had to implement the partial loading logic in both places.

Reply:
Perhaps related, I successfully ran these:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset(
    "activitynet-100",
    split="validation",
    max_samples=10,
)

dataset = foz.load_zoo_dataset(
    "activitynet-200",
    split="validation",
    classes=["Bathing dog", "Walking the dog"],
    max_duration=20,
    max_samples=4,
)
```

But then I tried (after exiting and reopening ipython):

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset(
    "activitynet-200",
    split="validation",
    classes=["Bathing dog", "Walking the dog"],
)
```

and got the following error:

```
~/dev/fiftyone/fiftyone/utils/activitynet.py in download_activitynet_split(dataset_dir, split, source_dir, classes, max_duration, copy_files, num_workers, shuffle, seed, max_samples, version)
    122             sample_ids,
    123             num_downloaded_samples,
--> 124         ) = manager.download_necessary_samples(download_config)
    125         num_samples = num_existing + num_downloaded_samples
    126

~/dev/fiftyone/fiftyone/utils/activitynet.py in download_necessary_samples(self, config)
    549             requested_num,
    550             num_workers,
--> 551             split,
    552         )
    553         requested_sample_ids.extend(downloaded_samples)

~/dev/fiftyone/fiftyone/utils/activitynet.py in _download_requested_samples(self, ids, class_samples, shuffle, requested_num, num_workers, split)
    585
    586             downloaded_ids = self._separate_versions_and_attempt_to_download(
--> 587                 ids, class_samples, requested_num, num_workers, split,
    588             )
    589

~/dev/fiftyone/fiftyone/utils/activitynet.py in _separate_versions_and_attempt_to_download(self, ids, samples_info, num_samples, num_workers, split)
    608                 a100_ids,
    609                 samples_info,
--> 610                 min(num_a100_ids, num_samples),
    611                 num_workers,
    612             )

TypeError: '<' not supported between instances of 'NoneType' and 'int'
```